src/robot/libraries/BuiltIn.py

```diff
@@ -1072,6 +1072,11 @@ def should_contain(self, container, item, msg=None, values=True,
         in Robot Framework 4.1.
         """
         orig_container = container
+        if isinstance(container, bytes) and isinstance(item, str):
+            try:
+                item = item.encode('latin-1')
+            except UnicodeEncodeError:
+                raise ValueError(f"{item} can not be encoded into bytes.")
```

Comment:
You added a test where the item is `1000` and it fails for a ValueError. Should we actually validate that, if the item is an integer, it is in range 0-255 and provide a nicer error than the current `byte must be in range(0, 256)` that Python gives us? That would also avoid the issue with the error possibly changing in some future Python version.